examples/files/hooks/useReducer.js

```diff
@@ -0,0 +1,41 @@
+import React, { useReducer } from 'react'
+
+const App = () => {
+  const reducer = (state, action) => {
+    if (action.type === 'cat') {
+      return 'You selected cat'
+    } else if (action.type === 'dog') {
+      return 'You selected dog'
+    }
+    return state
+  }
+  const [state, dispatch] = useReducer(reducer, 'No Pet Selected')
+  return (
+    <div>
+      <input
+        type="radio"
+        name="pet"
+        value="cat"
+        onClick={() => {
+          dispatch({ type: 'cat' })
```

Comment:
I think "cat" or "dog" would be the `value` or `payload` of an action, rather than the `type`.

This example is tricky since we want to keep it short, but also, `useReducer` is really only relevant when we have sub-values in our state.

I'd probably make state contain 2 different sub-values, e.g. `{ pet: "cat", color: "black" }`, and then dispatch actions like `{ type: "pet", value: "dog" }`.

We can possibly use `<select>` instead of radio buttons to save space.

It'd come out a bit longer, but, it'd be a more realistic example.